packages/@webex/plugin-meetings/src/meeting/index.ts

```diff
@@ -1386,6 +1393,74 @@ export default class Meeting extends StatelessWebexPlugin {
     this.callStateForMetrics.correlationId = correlationId;
   }
 
+  /**
+   * Set meeting info and trigger `MEETING_INFO_AVAILABLE` event
+   * @param {any} info
+   * @param {string} meetingLookupUrl lookup url
+   * @returns {void}
+   */
+  private setMeetingInfo(info, meetingLookupUrl) {
+    this.requiredCaptcha = null;
+    if (
+      this.passwordStatus === PASSWORD_STATUS.REQUIRED ||
+      this.passwordStatus === PASSWORD_STATUS.VERIFIED
+    ) {
+      this.passwordStatus = PASSWORD_STATUS.VERIFIED;
+    } else {
+      this.passwordStatus = PASSWORD_STATUS.NOT_REQUIRED;
+    }
+
+    this.parseMeetingInfo(info, this.destination);
+    this.meetingInfo = info ? {...info, meetingLookupUrl} : null;
+    this.meetingInfoFailureReason = MEETING_INFO_FAILURE_REASON.NONE;
+
+    Trigger.trigger(
+      this,
+      {
+        file: 'meetings',
+        function: 'fetchMeetingInfo',
+      },
+      EVENT_TRIGGERS.MEETING_INFO_AVAILABLE
+    );
+
+    this.updateMeetingActions();
+  }
+
+  public async injectMeetingInfo(info: any, fetchParams: FetchMeetingInfoParams): Promise<void> {
+    await this.prepForFetchMeetingInfo(fetchParams, 'injectMeetingInfo');
+
+    this.setMeetingInfo(info, undefined);
```

Comment:
I updated the JSDoc and set the default value for the `meetingLookupUrl` parameter to `undefined`

Reply:
My point wasn't that we shouldn't be providing `undefined` as an argument, but that we should allow for the lookup URL to be provided when calling `injectMeetingInfo`, so that externally we can state where the information came from when injecting it. For example, in guest join cases, it will come from a preJoin API. Knowing this helps us debug issues with meeting info. So my suggestion was effectively that we should provide an extra argument / add to options to provide this here and in the create method (where this is used). i.e.

```js
public async injectMeetingInfo(info: any, fetchParams: FetchMeetingInfoParams, meetingLookupUrl: string) {
```

Or something like that.

Reply:
I see you point now. 

I refactored it, so meetingLookUp url can passed from cantina if it is available